Source/WebCore/editing/cocoa/DataDetection.mm

```diff
@@ -113,7 +113,12 @@
     if (!view)
         return { };
 
+#if HAVE(SECURE_ACTION_CONTEXT)
+    auto actionContext = adoptNS([PAL::allocDDSecureActionContextInstance() init]);
+#else
     auto actionContext = adoptNS([PAL::allocDDActionContextInstance() init]);
+#endif
```

Comment:
You could do something similar (write an inline method) in `DataDetectorsSPI.h` to eliminate this code as well:

```
    inline WKDDActionContext PAL::allocWKDDActionContextInstance()
    {
#if HAVE(SECURE_ACTION_CONTEXT)
        return PAL::allocDDSecureActionContextInstance();
#else
        return PAL::allocDDActionContextInstance();
#endif
    }
```

Reply:
I like it.